site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,264 @@
+
+![Update Tenant Information message flow](../tenant_UpdateTenantSuccess.png)
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *update tenant* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name        | Mandatory | Location                 | Type     | Description |
+| :---------- | :-------: | :----------------------- | :------- | :---------- |
+| *subject*   | yes       | *properties*             | *string* | MUST be set to `update`. |
+
+The request message MUST include payload as defined in the [Tenant Format]({{< relref "#tenant-format" >}}) section below.
+ 
+**Response Message Format**
+
+A response to an *update tenant* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the tenant has been updated successfully. |
+| *404* | Not Found, there is no tenant identified with the given *tenant-id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Remove Tenant
+
+Clients use this command to *remove* all information about a tenant from Hono. Once a tenant is removed, clients can no longer use Hono to interact with any device of the tenant nor can they consume any more data produced by the devices of the tenant.
+
+This operation is *optional*, implementors of this API may provide other means for removing tenant information, e.g. a RESTful API or a configuration file.
+**Message Flow**
+
+The following sequence diagram illustrates the flow of messages involved in a *Client* removing a device.
+
+![Remove Tenant message flow](../tenant_RemoveTenantSuccess.png)
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a *remove tenant* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name        | Mandatory | Location                 | Type     | Description |
+| :---------- | :-------: | :----------------------- | :------- | :---------- |
+| *subject*   | yes       | *properties*             | *string* | MUST be set to `remove`. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+
+**Response Message Format**
+
+A response to a *remove tenant* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response may contain the following status codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the tenant has been removed successfully. |
+| *404* | Not Found, there is no tenant identified with the given *tenant-id*. |
```

Comment:
`... given *tenant_id*`
NB: the *tenant-id* property should be renamed to *tenant_id* to support JMS based clients.

Reply:
adapted